apps/examples/testcase/le_tc/filesystem/fs_main.c

```diff
@@ -1961,6 +1965,400 @@ static void libc_stdio_setvbuf_tc(void)
+* @scenario         Initializes a NULL stream. The initialized stream will return only EOF
+* @apicovered       lib_nullinstream
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_nullinstream_tc(void)
+{
+        union {
+                struct lib_instream_s nullinstream;
+                struct lib_meminstream_s meminstream;
+        } u;
+
+	FAR char buf[BUFLEN];
+	FAR struct lib_instream_s *stream;
+
+	lib_meminstream((FAR struct lib_meminstream_s *)&u.meminstream, buf, BUFLEN);
+	TC_ASSERT_EQ("lib_meminstream", u.meminstream.buffer, (FAR char *)(buf));
+
+	lib_nullinstream(&u.nullinstream);
+	stream = &u.nullinstream;
+	TC_ASSERT_EQ("lib_nullinstream", u.nullinstream.nget, 0);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         libc_stdio_nulloutstream_tc
+* @brief            Initializes a NULL stream. The initialized stream will write all data to the bit-bucket
+* @scenario         Initializes a NULL stream. The initialized stream will write all data to the bit-bucket
+* @apicovered       lib_nulloutstream
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_nulloutstream_tc(void)
+{
+        union {
+                struct lib_outstream_s nulloutstream;
+                struct lib_memoutstream_s memoutstream;
+        } u;
+
+	FAR char buf[BUFLEN];
+	FAR struct lib_outstream_s *stream;
+
+	lib_memoutstream((FAR struct lib_memoutstream_s *)&u.memoutstream, buf, BUFLEN);
+	TC_ASSERT_EQ("lib_memoutstream", u.memoutstream.buffer, (FAR char *)(buf));
+
+	lib_nulloutstream(&u.nulloutstream);
+	stream = &u.nulloutstream;
+	TC_ASSERT_EQ("lib_nulloutstream", u.nulloutstream.nput, 0);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         libc_stdio_rawinstream_tc
+* @brief            Initializes a stream for use with a file descriptor
+* @scenario         Initializes a stream for use with a file descriptor
+* @apicovered       lib_rawinstream
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_rawinstream_tc(void)
+{
+	FILE *fd;
+	char *filename = VFS_FILE_PATH;
+
+	struct lib_rawinstream_s rawinstream;
+
+	fd = fopen(filename, "w");
+	TC_ASSERT("fopen", fd);
+
+	lib_rawinstream((FAR struct lib_rawinstream_s *)&rawinstream, (int)fd);
+	TC_ASSERT_EQ_CLEANUP("lib_rawinstream", rawinstream.public.nget, 0, fclose(fd));
+
+	fclose(fd);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         libc_stdio_rawoutstream_tc
+* @brief            Initializes a stream for use with a file descriptor
+* @scenario         Initializes a stream for use with a file descriptor
+* @apicovered       lib_rawoutstream
+* @precondition     NA
+* @postcondition    NA
+*/
+static void libc_stdio_rawoutstream_tc(void)
+{
+	FILE *fd;
+	char *filename = VFS_FILE_PATH;
+
+	struct lib_rawoutstream_s rawoutstream;
+
+	fd = fopen(filename, "w");
+	TC_ASSERT("fopen", fd);
+	fclose(fd);
+
+	lib_rawoutstream((FAR struct lib_rawoutstream_s *)&rawoutstream, (int)fd);
```

Comment:
in line 2158, you closed the fd, but you used that fd in this line.
is it okay?

Reply:
no, will take care of that, left out mistakenly